contracts/Remittance.sol

```diff
@@ -0,0 +1,98 @@
+pragma solidity 0.5.10;
+
+import "./SafeMath.sol";
+import "./Stoppable.sol";
+
+contract Remittance is Stoppable {
+    using SafeMath for uint;
+    
+    struct Order {
+        uint deadline;
+        address emitter;
+        uint amount;
+    }
+    
+    uint constant TX_FEES = 2000;
+
+    mapping(bytes32 => Order) public orders;
+    mapping(address => uint) public fees;
+    
+    event LogNewOrder(address indexed emitter, uint amount, bytes32 hashedOTP);
+    event LogTakeFees(address indexed emitter, uint amount);
+    event LogCancelOrder(address indexed emitter, uint amount, bytes32 hashedOTP);
+    event LogWithdraw(address indexed emitter, uint amount);
+    
+    constructor(bool state) Stoppable(state) public {}
+    
+    function hashOTP(address exchange, bytes32 password) public view returns(bytes32 hash) {
+        require(exchange != address(0) && password != 0, "Both exchange and password must be set");
+        hash = keccak256(abi.encodePacked(address(this), exchange, password));
+    }
+    
+    function newOrder(bytes32 hashedOTP, uint delay) _onlyIfRunning public payable returns(bool success) {
+        require(hashedOTP != 0 && msg.value > 0, "An error occured. Ensure that the secret is set and that your transaction value is above 0");
+        require(delay > 0 days, "Delay should be above 0 day");
+        require(orders[hashedOTP].emitter == address(0), "Password already used");
+        
+        uint amount = msg.value.sub(TX_FEES);
+        address owner = getOwner();
+        
+        fees[owner] = fees[owner].add(TX_FEES);
+        emit LogTakeFees(msg.sender, TX_FEES);
+        
+        emit LogNewOrder(msg.sender, amount, hashedOTP);
+        orders[hashedOTP] = Order({
+            deadline: now.add(delay),
```

Comment:
We did not discuss it but you have to know that the timestamp is not very accurate, but good enough for this use case.